indexer/src/test/java/uk/ac/ebi/uniprot/indexer/uniprotkb/UniProtKBJobIT.java

```diff
@@ -54,18 +57,46 @@ void testUniProtKBIndexingJob() throws Exception {
         BatchStatus status = jobExecution.getStatus();
         assertThat(status, is(BatchStatus.COMPLETED));
 
-        Page<UniProtDocument> response = template
-                .query(SolrCollection.uniprot.name(), new SimpleQuery("*:*"), UniProtDocument.class);
+        Collection<StepExecution> stepExecutions = jobExecution.getStepExecutions();
+        assertThat(stepExecutions, hasSize(2));
+
+        checkUniProtKBIndexingStep(jobExecution, stepExecutions);
+        checkSuggestionIndexingStep(stepExecutions);
+    }
+
+    private void checkSuggestionIndexingStep(Collection<StepExecution> stepExecutions) {
+        StepExecution suggestionIndexingStep = stepExecutions.stream()
+                .filter(step -> step.getStepName().equals(SUGGESTIONS_INDEX_STEP))
+                .collect(Collectors.toList()).get(0);
+
+        assertThat(suggestionIndexingStep.getReadCount(), is(greaterThan(0)));
+
+        int reportedWriteCount = suggestionIndexingStep.getWriteCount();
+        assertThat(reportedWriteCount, is(greaterThan(0)));
+        assertThat(suggestionIndexingStep.getSkipCount(), is(0));
+        assertThat(suggestionIndexingStep.getFailureExceptions(), hasSize(0));
+
+
+        Page<SuggestDocument> response = template
+                .query(SolrCollection.suggest.name(), new SimpleQuery("*:*"), SuggestDocument.class);
         assertThat(response, is(notNullValue()));
-        assertThat(response.getTotalElements(), is(5L));
+        assertThat(response.getTotalElements(), is((long)reportedWriteCount));
+    }
 
-        StepExecution indexingStep = jobExecution.getStepExecutions().stream()
+    private void checkUniProtKBIndexingStep(JobExecution jobExecution, Collection<StepExecution> stepExecutions) {
+        StepExecution kbIndexingStep = stepExecutions.stream()
                 .filter(step -> step.getStepName().equals(UNIPROTKB_INDEX_STEP))
                 .collect(Collectors.toList()).get(0);
 
-        assertThat(indexingStep.getReadCount(), is(5));
+        assertThat(kbIndexingStep.getReadCount(), is(5));
         checkWriteCount(jobExecution, Constants.INDEX_FAILED_ENTRIES_COUNT_KEY, 0);
         checkWriteCount(jobExecution, Constants.INDEX_WRITTEN_ENTRIES_COUNT_KEY, 5);
+
+        Page<UniProtDocument> response = template
```

Comment:
Do you think it is worst assert some information about the document as well?

Reply:
done, hopefully?!